### PR TITLE
GH-409: Admin container restructured to allow specific permissions, c…

### PR DESCRIPTION
**Pull request by Midnite1212**
Old PR got deleted after cleaning unnecessary commits

### CODE REVIEW

File: ui/src/components/MainContainer.js [hunk 1]
```diff
@@ -168,13 +168,13 @@ const MainContainer = () => {
         <PrivateRoute
           exact
           path="/clear-cache/"
-          permissions={['admin', 'stewardship']}
+          permissions={['t3ch']}
```
Comment (albertparedandan) [inline ui/src/components/MainContainer.js:171]: should be tech and admin for now

File: ui/src/components/MainContainer.js [hunk 2]
```diff
@@ -210,7 +210,7 @@ const MainContainer = () => {
         <PrivateRoute
           exact
           path="/admin/:pageName"
-          permissions={['admin', 'stewardship']}
+          permissions={['admin', 'stewardship', 't3ch']}
```
Comment (albertparedandan) [inline ui/src/components/MainContainer.js:213]: Hmm, I don't think this is the best implementation for admin pages. As of right now it means we can't permission each admin pages individually (which we want to)

File: ui/src/components/AdminContainer.js [hunk 3]
```diff
@@ -28,11 +29,51 @@ const AdminContainer = (props) => {
       >
         {
           {
-            home: <AdminHome {...props} />,
-            users: <AdminUser {...props} />,
-            forms: <FormManager {...props} />,
-            giving: <AdminGiving {...props} />,
-            logout: <AdminLogout {...props} />,
+            home: (
+              <PrivateRoute
```
Comment (albertparedandan) [inline ui/src/components/AdminContainer.js:33]: Why are we putting a privateroute inside a privateroute?

**Review by albertparedandan — CHANGES_REQUESTED** [review]

**Review by albertparedandan — CHANGES_REQUESTED** [review]

File: ui/src/components/AdminContainer.js [hunk 4]
```diff
@@ -18,6 +19,30 @@ const AdminContainer = (props) => {
   const variants = useBreakpointValue({ base: smVariant, md: mdVariant });
   const toggleSidebar = () => setSidebarOpen(!isSidebarOpen);
 
+  const allowHome = () => {
```
[the remark this answers is not in the document]
Comment (Midnite1212) [reply ui/src/components/AdminContainer.js:22]: If the adminContainer is combined with main, how about the sidebar in admin container? Shod I just move it to main too? Or..?

File: ui/src/components/AdminContainer.js [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
-import AdminUser from "./admin/users/AdminUser";
-import FormManager from "./forms/FormManager";
-import AdminGiving from "./admin/AdminGiving";
-import SidebarWithHeader from "./admin/navigation/Sidebar";
-import { useState } from "react";
-import AdminLogout from "./admin/AdminLogout";
-
-const smVariant = { navigation: "drawer", navigationButton: true };
-const mdVariant = { navigation: "sidebar", navigationButton: false };
+import React from 'react';
```
Comment (albertparedandan) [inline ui/src/components/AdminContainer.js:1]: is there a use for adminContainer.js file? if not, can we delete it

File: ui/src/components/MainContainer.js [hunk 6]
```diff
@@ -23,10 +23,15 @@ import ConnectFloatButton from './connect-float/ConnectFloatButton';
 import UserFormContainer from './forms/UserFormContainer';
 import NoMatch from './errors/NoMatch';
 import AdminLoginContainer from './admin/AdminLoginContainer';
-import AdminContainer from './AdminContainer';
+//import AdminContainer from './AdminContainer';
```
Comment (albertparedandan) [inline ui/src/components/MainContainer.js:26]: remove

**Review by albertparedandan — CHANGES_REQUESTED** [review]

File: ui/src/components/MainContainer.js [hunk 7]
```diff
@@ -168,51 +173,54 @@ const MainContainer = () => {
...
-          permissions={["admin", "stewardship"]}
-          component={AdminForm}
-        />
-        <PrivateRoute
-          exact
-          path="/admin/giving"
-          permissions={["stewardship"]}
-          component={AdminGiving}
-        />  */}
         <PrivateRoute
```
Comment (albertparedandan) [inline ui/src/components/MainContainer.js:185]: why are there 2?

**Review by albertparedandan — CHANGES_REQUESTED** [review]

File: ui/src/components/MainContainer.js (hunk 7 above)
Comment (Midnite1212) [reply ui/src/components/MainContainer.js:185]: cuz the logout will redirect to admin/logout instead of /admin, should I just change the logout.js then?

**Review by albertparedandan — APPROVED** [review]

**Pull request merged**